tests/components/cover/test_demo.py

```diff
@@ -0,0 +1,92 @@
+"""The tests for the Demo cover platform."""
+import unittest
+import homeassistant.util.dt as dt_util
+
+from homeassistant.components.cover import demo
+from tests.common import fire_time_changed, get_test_home_assistant
+
+
+class TestCoverDemo(unittest.TestCase):
+    """Test the Demo cover."""
+
+    def setUp(self):  # pylint: disable=invalid-name
+        """Setup things to be run when tests are started."""
+        self.hass = get_test_home_assistant()
+
+    def tearDown(self):  # pylint: disable=invalid-name
+        """Stop down everything that was started."""
+        self.hass.stop()
+
+    def test_close_cover(self):
+        """Test closing the cover."""
+        entity = demo.DemoCover(self.hass, 'test', 100, None)
+        entity.close_cover()
+
+        fire_time_changed(self.hass, dt_util.utcnow())
+        self.hass.pool.block_till_done()
+        self.assertEqual(90, entity.current_cover_position)
+
+    def test_open_cover(self):
+        """Test opening the cover."""
+        entity = demo.DemoCover(self.hass, 'test', 0, None)
+        entity.open_cover()
+
+        fire_time_changed(self.hass, dt_util.utcnow())
+        self.hass.pool.block_till_done()
+        self.assertEqual(10, entity.current_cover_position)
+
+    def test_set_cover_position(self):
+        """Test moving the cover to a specific position."""
+        entity = demo.DemoCover(self.hass, 'test', 0, None)
+        entity.set_cover_position(10)
+
+        fire_time_changed(self.hass, dt_util.utcnow())
```

Comment:
I meant the block and the assertion and the fire time changed are literally
copy pasta'd - a quick helper method that takes the expected position could
reduce the code by around 15 lines.

On Aug 21, 2016 10:29 AM, "Paulus Schoutsen" <notifications@github.com>
wrote:

> In tests/components/cover/test_demo.py
> <https://github.com/home-assistant/home-assistant/pull/2891#discussion_r75598037>
> :
>
> > +
> > +    def test_open_cover(self):
> > +        """Test opening the cover."""
> > +        entity = demo.DemoCover(self.hass, 'test', 0, None)
> > +        entity.open_cover()
> > +
> > +        fire_time_changed(self.hass, dt_util.utcnow())
> > +        self.hass.pool.block_till_done()
> > +        self.assertEqual(10, entity.current_cover_position)
> > +
> > +    def test_set_cover_position(self):
> > +        """Test moving the cover to a specific position."""
> > +        entity = demo.DemoCover(self.hass, 'test', 0, None)
> > +        entity.set_cover_position(10)
> > +
> > +        fire_time_changed(self.hass, dt_util.utcnow())
>
> fire_time_changed is already the helper method ? :)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/2891/files/0e9ae1b1074b7511838688de160ab4875085a86d#r75598037>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC2fZYx_NT5nmp_XYJls63aOdzRIeAt7ks5qiHz9gaJpZM4Jo0hM>
> .
>